### PR TITLE
docs: expand rockcraft.yaml reference

### PR DESCRIPTION
**Pull request by tigarmo**
*Review note*: The first commit re-organizes the "reference" section into multiple documents, but with no changes in content. The second commit is the one that rewrites the rockcraft.yaml reference.

### CODE REVIEW

File: docs/reference/commands.rst [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+stage
+^^^^^
+Stages built artifacts into a common staging area, for each part.
+
+prime
+^^^^^
+Prepare, for each part, the final payload to be packed as a ROCK, performing
+additional processing and adding metadata files.
+
+pack
```
[the remark this answers is not in the document]
Comment (sergiusens) [reply docs/reference/commands.rst:39]: we should think about an rst formatter to autogen our docs (@dboddie) :-)

File: docs/reference/parts.rst [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+Example
+.......
+
+.. _chisel-example:
+
+.. code-block:: yaml
+
+  parts:
+    chisel-openssl-binaries-only:
```
Comment (cjdcordeiro) [inline docs/reference/parts.rst:41]: this part's name is a bit misleading

**Review by cjdcordeiro — APPROVED** [review]
nice thanks. apart from a nitpick, what @sergiusens showed is quite cool and would make the `commands.rst` much more useful, cause tbh, atm, they don't help much.

**Comment by tigarmo** [comment]
I opened #293 so that we can improve the reference for the commands

File: docs/reference/parts.rst (hunk 2 above)
Comment (tigarmo) [reply docs/reference/parts.rst]: That bit was already there but in the spirit of Friday I updated it to `chisel-openssl` ;)

File: docs/reference/commands.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (tigarmo) [reply docs/reference/commands.rst:39]: I think we can use pandoc to convert the markdown into rst

**Comment by tigarmo** [comment]
Rebased to keep the merge as two separate commits

**Review by dboddie — APPROVED** [review]
Looks good to go. :)

**Pull request merged**